### PR TITLE
[docs]: update comments for github_branch_protection in case of push_restrictions

### PR DESCRIPTION
**Pull request by Nmishin**
Resolves https://github.com/integrations/terraform-provider-github/issues/1841

----

### Before the change?

* Docks have a wrong comments

### After the change?

* Docks have a right comments  about push_restrictions, and also note for the right access rights.

### Pull request checklist
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)

### Does this introduce a breaking change?

Please see our docs on [breaking changes](https://github.com/octokit/.github/blob/master/community/breaking_changes.md) to help!

- [ ] Yes
- [x] No

----

### CODE REVIEW

**Comment by macetw** [comment]
Awesome, Mishin, but I'd like to suggest your examples include a mix of team/user. This was the original problem with the documentation, that it was shown as a bug. But In your examples, it isn't really demonstrating the way this can be overcome.

Thanks for working this!!

File: website/docs/r/branch_protection.html.markdown [hunk 1]
```diff
@@ -47,16 +49,18 @@ resource "github_branch_protection" "example" {
     data.github_user.example.node_id,
     "/exampleuser",
     "exampleorganization/exampleteam",
-    # limited to a list of one type of restriction (user, team, app)
+    # you can have more than one type of restriction (teams + users)
```
Comment (macetw) [inline website/docs/r/branch_protection.html.markdown:52]:
```suggestion
    # you can have more than one type of restriction (teams + users), but if
    # you use more than one type, use node_ids of the given user or team.
```

File: website/docs/r/branch_protection.html.markdown [hunk 2]
```diff
@@ -47,16 +49,18 @@ resource "github_branch_protection" "example" {
...
     # github_team.example.node_id
+    # github_team.example-2.node_id
```
Comment (macetw) [inline website/docs/r/branch_protection.html.markdown:54]:
```suggestion
    # github_user.example-2.node_id
```

**Comment by macetw** [comment]
Added suggestions. Thanks again, Mishin!

**Comment by Nmishin** [comment]
macetw, thank you for review! 
I added your commits, please check, it looks better now?

**Review by macetw — APPROVED** [review]
Looks goo, Mishin!

**Review by nickfloyd — APPROVED** [review]
Hey @Nmishin thank you for these contributions! ❤️

**Pull request merged**